api/api.go

```diff
@@ -129,8 +136,16 @@ func (s *authedRequester) Do(r *Request) *Response {
 
 	theurl := s.baseurl
 	theurl.Path += r.Endpoint
-	theurl.RawQuery = urlParams.Encode()
-	req, err := http.NewRequest("GET", theurl.String(), nil)
+	if r.ReqType == "POST" {
+		r.Body.WriteString(urlParams.Encode())
+	} else {
+		theurl.RawQuery = urlParams.Encode()
+	}
+	req, err := http.NewRequest(r.ReqType, theurl.String(), bytes.NewReader(r.Body.Bytes()))
+	if r.ReqType == "POST" {
+		req.Header.Set("Content-Type", "application/x-www-form-urlencoded; param=value")
```

Comment:
It's not clear from the code why this is needed?  Not sure whether this needs a comment, or maybe some refactoring to make this obvious.

Reply:
This is indeed needed from what I understand